app/models/document.rb

```diff
@@ -0,0 +1,55 @@
+class Document
+  include ActiveModel::Model
+
+  attr_accessor :content_id, :title, :summary, :body, :format_specific_fields, :state, :bulk_published, :minor_update, :change_note
+
+  def initialize(params = {}, format_specific_fields = [])
```

Comment:
I'm not sure why format_specific_fields needs to be a parameter to the initialize method. They are available on the object already, you can just refer to them directly. There probably needs to be a default declaration of `FORMAT_SPECIFIC_FIELDS = []` on the base Document class which is overridden as you have done in CmaCase.